src/WebJobs.Script.WebHost/Configuration/HostEasyAuthOptions.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Configuration
+{
+    public class HostEasyAuthOptions
+    {
+        public bool SiteAuthEnabled { get; set; }
+
+        public string SiteAuthClientId { get; set; }
+
+        public IConfiguration Configuration { get; set; }
```

Comment:
Configuration is not needed here. In HostEasyAuthOptionsSetup, extract all the properties needed from configuration

Reply:
There is a huge amount of config needed by EasyAuth, in a very nested structure (I would say 30-40 fields), and as we add new features, we will likely need more. By passing us the IConfiguration directly, it allows us to keep this configuration extraction all in the EasyAuth codebase, instead of duplicating it here.